website/docs/using-odo/create-component.md

```diff
@@ -3,4 +3,139 @@ title: Create Component
+
+In this guide, we are going to create a Spring Boot component to deploy the [Spring Boot petclinic](https://github.com/spring-projects/spring-petclinic) project to a Kubernetes cluster.
+
+Let's clone the project first:
+```shell
+git clone https://github.com/spring-projects/spring-petclinic
+cd spring-petclinic
+```
+
+Next, let's create a project <!-- add link to project command reference here --> on the Kubernetes cluster in which we will be creating our component:
+```shell
+odo project create myproject
+```
+Alternatively, you could also use one of the existing projects on the cluster:
+```shell
+odo project list
+odo project set <project-name>  # replace <project-name> with a valid value from the list
+```
+
+odo supports interactive and non-interactive ways of creating a component.
+
+## Interactively creating a component
+
+To interactively create a component, `cd` into the cloned petclinic project (already done if you copy-pasted the command above) and execute:
+```shell
+odo create
+```
+You will be prompted with a few questions one after the another. Let's go through each one of them to create a component.
+
+1. First question is about selecting the component type:
+    ```shell
+    odo create
+    ? Which devfile component type do you wish to create  [Use arrows to move, enter to select, type to filter]
+    > java-maven
+    java-maven
+    java-openliberty
+    java-openliberty
+    java-quarkus
+    java-quarkus
+    java-springboot
+    ```
+   By default, `java-maven` is selected for us. Since this is a Spring Boot application, we should be selecting `java-springboot`. 
+
+    We can either scroll down to `java-springboot` using the arrow key, or start typing `spring` on the prompt. Typing `spring` will lead to odo filtering the component type based on your input.
+
+2. Next, odo asks you to name the component:
+    ```shell
+    odo create                
+    ? Which devfile component type do you wish to create java-springboot
+    ? What do you wish to name the new devfile component (java-springboot) petclinic
+    ```
+    Let's name it `petclinic`.
+
+3. Next, odo asks you for the project in which you would like to create the component. Use the project `myproject` that we created earlier or the one you had set using `odo project set` command
+   ```shell
+   odo create
+   ? Which devfile component type do you wish to create java-springboot
+   ? What do you wish to name the new devfile component java-springboot
+   ? What project do you want the devfile component to be created in myproject
+   ```
+   Now you will have a `devfile.yaml` in your current working directory. But odo is just not done asking you questions yet.
+4. Lastly, odo asks you if you would like to download a starter project. Since we already cloned the petclinic project, we answer in No by typing `n` and hitting the return key:
```

Comment:
> Lastly, odo asks you if you would like to download a starter project.

Asking from a reader's perspective: What is a starter project?

I had this question a lot of times when I was getting started with odo; it might help to give `starter project` an alias, say `example project` in the brackets. Or better yet, add the hyperlink to the starter project section in the doc.

> hitting the return key

Return key? Shouldn't it be the enter key?

Reply:
> Asking from a reader's perspective: What is a starter project?
> 
> I had this question a lot of times when I was getting started with odo; it might help to give `starter project` an alias, say `example project` in the brackets. Or better yet, add the hyperlink to the starter project section in the doc.

Thanks for the catch! I had put a reference in the beginning of the doc. So I didn't put it here. But then I removed that statement and didn't remember to put it here. Updated. PTAL.

> Return key? Shouldn't it be the enter key?

They are the same. I have observed "Return" used more widely in docs.